c/tskit/trees.c

```diff
@@ -4789,20 +4771,21 @@ check_kc_distance_tree_sequence_inputs(tsk_treeseq_t *self, tsk_treeseq_t *other
 }
 
 static void
-update_kc_pairs_with_leaf(tsk_tree_t *self, kc_vectors *kc, tsk_id_t leaf,
+update_kc_pair_with_sample(tsk_tree_t *self, kc_vectors *kc, tsk_id_t sample,
     tsk_size_t *depths, double root_time)
 {
     tsk_id_t c, p, sib;
     double time;
     tsk_size_t depth;
     double *times = self->tree_sequence->tables->nodes.time;
 
-    for (c = leaf, p = self->parent[leaf]; p != TSK_NULL; c = p, p = self->parent[p]) {
+    for (c = sample, p = self->parent[sample]; p != TSK_NULL;
```

Comment:
Eek, didn't notice the comma operator before. I'd prefer to avoid this - presumably it's the same thing if we maintain the c variable within the loop body?